### PR TITLE
Show Feed to user

### LINKED ISSUE #514: Show Feed to user
Add feed {user}/feed/feed.atom in this https://github.com/tarasglek/chatcraft.org/issues/425, we need a place to show it to user.

### PR DESCRIPTION
**Pull request by WangGithub0**
Add feed in 2 places: 
 **1. Add share feed button on header;** 
 
image

 **2. In shareModal**
 
image

closes #514

### CODE REVIEW

**Comment by humphd** [comment]
@WangGithub0 can you rebase for the fix to Header.tsx?

File: src/components/Header.tsx [hunk 1]
```diff
@@ -69,6 +74,30 @@ function Header({ chatId, inputPromptRef, searchText, onToggleSidebar }: HeaderP
...
+      error({
+        title: "Failed to Share Message",
+        message: "Can't share message because user is not logged in",
+      });
+      return;
+    }
+    try {
+      const UserFeedUrl = `https://chatcraft.org/api/share/${user.username}/feed.atom`;
+      info({
+        title: "Copy Shared Chats Feed URL Successfully",
```
Comment (humphd) [inline src/components/Header.tsx:88]: "Copy" should be "Copied"

File: src/components/ShareModal.tsx [hunk 2]
```diff
@@ -53,6 +54,9 @@ function AuthenticatedForm({ chat, user }: AuthenticatedForm) {
         throw new Error("Unable to create Share URL");
       }
       setUrl(url);
+      const baseUrl = url.split("/api/")[0];
```
Comment (humphd) [inline src/components/ShareModal.tsx:57]: This kind of thing worries me (hacking on URLs as strings).  Use `new URL()` and then pull stuff out of the pathname with that.  Same for building new URLs.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/components/Header.tsx [hunk 3]
```diff
@@ -69,6 +74,30 @@ function Header({ chatId, inputPromptRef, searchText, onToggleSidebar }: HeaderP
...
+  const handleCopyFeedUrl = useCallback(async () => {
+    if (!user) {
+      error({
+        title: "Failed to Share Message",
+        message: "Can't share message because user is not logged in",
+      });
+      return;
+    }
+    try {
+      const UserFeedUrl = `https://chatcraft.org/api/share/${user.username}/feed.atom`;
```
Comment (Amnish04) [inline src/components/Header.tsx:86]: Not a big issue, but this should be camel cased by convention?

File: src/components/ShareModal.tsx [hunk 4]
```diff
@@ -53,6 +54,9 @@ function AuthenticatedForm({ chat, user }: AuthenticatedForm) {
         throw new Error("Unable to create Share URL");
       }
       setUrl(url);
+      const parsedUrl = new URL(url);
+      const newUserFeedUrl = `${parsedUrl.origin}/api/share/${user.username}/feed.atom`;
```
Comment (humphd) [inline src/components/ShareModal.tsx:58]: This is good.  One other way you can do this is to create a URL for a path given another URL:

```ts
const url = new URL('/some/path', parsedUrl.origin);
```

**Review by humphd — APPROVED** [review]

**Pull request merged**